--faulthandler-timeout on Windows/Py27 CI runs blows up

Issue seems to be identical to #8, was thinking . about commenting there but given that was 4 years ago, I thought I would open my own issue.

On azure-pipelines I . call pytest in the following fashion

```
      python -m pytest -sv \
        --junitxml=junit/test-results.xml \
        --cov pyqtgraph --cov-report=xml --cov-report=html \
        --faulthandler-timeout=60
```

When I make this call on my Windows/Python2.7 runs I get the following error:

```
========================== Starting Command Output ===========================
[command]"C:\Program Files\Git\bin\bash.exe" --noprofile --norc /d/a/_temp/a5488e34-446c-41fb-9365-433e149de497.sh
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "C:\Miniconda\envs\test-environment-2.7\lib\site-packages\_pytest\main.py", line 201, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "C:\Miniconda\envs\test-environment-2.7\lib\site-packages\_pytest\config\__init__.py", line 668, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "C:\Miniconda\envs\test-environment-2.7\lib\site-packages\pluggy\hooks.py", line 311, in call_historic
INTERNALERROR>     res = self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "C:\Miniconda\envs\test-environment-2.7\lib\site-packages\pluggy\manager.py", line 87, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "C:\Miniconda\envs\test-environment-2.7\lib\site-packages\pluggy\manager.py", line 81, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "C:\Miniconda\envs\test-environment-2.7\lib\site-packages\pluggy\callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "C:\Miniconda\envs\test-environment-2.7\lib\site-packages\pluggy\callers.py", line 81, in get_result
INTERNALERROR>     _reraise(*ex)  # noqa
INTERNALERROR>   File "C:\Miniconda\envs\test-environment-2.7\lib\site-packages\pluggy\callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "C:\Miniconda\envs\test-environment-2.7\lib\site-packages\pytest_faulthandler.py", line 40, in pytest_configure
INTERNALERROR>     config.warn(code='C1', message=message)
INTERNALERROR> AttributeError: 'Config' object has no attribute 'warn'
##[error]Bash exited with code '3'.
```

For all the other python/OS configurations besides PyQt4/PySide1 - Windows - Python2.7 it runs just fine.

--- a/.gitignore
+++ b/.gitignore
@@ -20,7 +20,7 @@ sdist/
 var/
 *.egg-info/
 .installed.cfg
-*.egg
+*.eggs
 
 # PyInstaller
 #  Usually these files are written by a python script from a template

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,12 +1,13 @@
 language: python
 python:
-  - 2.7
-  - 3.4
-  - 3.5
-  - 3.6
-  - nightly
-  - pypy2.7-5.10.0
-  - pypy3.5-5.10.1
+  - '2.7'
+  - '3.4'
+  - '3.5'
+  - '3.6'
+  - '3.7'
+  - 'pypy'
+  - 'pypy3'
+  - '3.8-dev'
 install:
   - pip install tox-travis
 script:

--- a/.travis.yml
+++ b/.travis.yml
@@ -14,6 +14,13 @@ install:
   - pip install tox-travis
 script:
   - tox
+
+stages:
+- name: test
+  if: repo = pytest-dev/pytest-faulthandler AND tag IS NOT present
+- name: deploy
+  if: repo = pytest-dev/pytest-faulthandler AND tag IS present
+
 jobs:
   include:
     - stage: test

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,4 +1,5 @@
 language: python
+dist: xenial
 python:
   - '2.7'
   - '3.4'
@@ -9,6 +10,7 @@ python:
   - 'pypy3'
   - '3.8-dev'
 install:
+  - pip install --upgrade pip
   - pip install tox-travis
 script:
   - tox

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,3 +1,8 @@
+1.6.0
+-----
+
+* pytest 4.0+ is now required.
+
 1.5.0
 -----
 

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -1,7 +1,12 @@
 1.6.0
 -----
 
+* The warning about fault-handler timeout not being supported in some platforms
+  has been moved to the section header (`#32`_).
+
 * pytest 4.0+ is now required.
+
+.. _#32: https://github.com/pytest-dev/pytest-faulthandler/issues/32
 
 1.5.0
 -----

--- a/README.rst
+++ b/README.rst
@@ -43,7 +43,7 @@ Requirements
 
 * Python 2.7+, Python 3.4+
 * pytest
-* faulthandler
+* faulthandler (Python 2.7)
 
 
 Install

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,4 +1,5 @@
 install:
+  - C:\Python36\python -m pip install --upgrade pip
   - C:\Python36\python -m pip install tox
 
 build: false  # Not a C# project

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,6 +1,6 @@
 install:
   - C:\Python36\python -m pip install --upgrade pip
-  - C:\Python36\python -m pip install tox
+  - C:\Python36\python -m pip install --upgrade tox virtualenv
 
 build: false  # Not a C# project
 

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -6,3 +6,5 @@ build: false  # Not a C# project
 
 test_script:
   - C:\Python36\scripts\tox --skip-missing-interpreters
+
+skip_tags: true

--- a/pytest_faulthandler.py
+++ b/pytest_faulthandler.py
@@ -49,7 +49,7 @@ def timeout_support_available():
         hasattr(faulthandler, 'cancel_dump_traceback_later')
 
 
-@pytest.mark.hookwrapper
+@pytest.hookimpl(hookwrapper=True)
 def pytest_runtest_protocol(item):
     enabled = item.config.getoption('fault_handler')
     timeout = item.config.getoption('fault_handler_timeout')

--- a/pytest_faulthandler.py
+++ b/pytest_faulthandler.py
@@ -1,7 +1,9 @@
-import sys
-import os
-import pytest
 import io
+import os
+import sys
+import warnings
+
+import pytest
 
 
 def pytest_addoption(parser):
@@ -32,12 +34,12 @@ def pytest_configure(config):
         config.fault_handler_stderr = os.fdopen(stderr_fd_copy, 'w')
         faulthandler.enable(config.fault_handler_stderr)
         # we never disable faulthandler after it was enabled, see #3
-        
-        if config.getoption('fault_handler_timeout') > 0:
-            if not timeout_support_available():
-                message = 'faulthandler timeout support not available on ' \
-                          'this platform'
-                config.warn(code='C1', message=message)
+
+
+def pytest_report_header(config):
+    if config.getoption('fault_handler_timeout') > 0:
+        if not timeout_support_available():
+            return 'faulthandler: timeout support not available on this platform'
 
 
 def timeout_support_available():

--- a/setup.py
+++ b/setup.py
@@ -10,7 +10,7 @@ setup(
     py_modules=['pytest_faulthandler'],
     url='https://github.com/pytest-dev/pytest-faulthandler',
     license='MIT',
-    install_requires=['pytest>=2.6'],
+    install_requires=['pytest>=4.0'],
     test_requires=['pytest-mock>=0.6'],
     setup_requires=['setuptools_scm'],
     author='Bruno Oliveira',

--- a/test_pytest_faulthandler.py
+++ b/test_pytest_faulthandler.py
@@ -93,9 +93,7 @@ def test_timeout_not_available(testdir):
     ''')
     result = testdir.runpytest('--faulthandler-timeout=5', '-rw')
     result.stdout.fnmatch_lines([
-        '*warnings summary*',
-        "*faulthandler timeout support not available on this platform*",
-        '*= 1 passed, 1 warnings in *',
+        "faulthandler: timeout support not available on this platform",
     ])
     assert result.ret == 0
 

--- a/tox.ini
+++ b/tox.ini
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{27,34,35,36},pypy,pypy3
+envlist = py{27,34,35,36,37},pypy,pypy3
 
 [testenv]
 commands = py.test test_pytest_faulthandler.py

